### PR TITLE
webrender: render colored and styled text

### PR DESCRIPTION
**Pull request by harryfei**

### CODE REVIEW

File: rust_src/crates/lisp/src/lisp.rs [hunk 1]
```diff
@@ -99,8 +99,8 @@ impl From<usize> for LispObject {
 }
 
 impl From<i32> for LispObject {
-    fn from(o: i32) -> Self {
-        LispObject::from_C(o as EmacsInt)
+    fn from(v: i32) -> Self {
+        Self::from_fixnum(EmacsInt::from(v))
```
Comment (DavidDeSimone) [inline rust_src/crates/lisp/src/lisp.rs:103]: This is a good change. I'll test that it doesn't effect JS, but it shouldn't

File: rust_src/src/webrender_backend/term.rs [hunk 2]
```diff
@@ -235,6 +332,37 @@ extern "C" fn frame_visible_invisible(frame: *mut Lisp_Frame, is_visible: bool)
     }
 }
 
+extern "C" fn set_background_color(f: *mut Lisp_Frame, arg: LispObject, _old_val: LispObject) {
```
Comment (brotzeit) [inline rust_src/src/webrender_backend/term.rs:335]: Can we use `mut frame: LispFrameRef` ?

Comment (harryfei) [reply rust_src/src/webrender_backend/term.rs:335]: Unfortunately, we can't.
This function is referenced in this below code. So the type checker need it's signature to be current one.

```rust
    let handlers: [frame_parm_handler; 47] = [
        None,
        None,
        None,
        None,
        None,
        Some(set_background_color),
        None,
        None,
        None,
        None,
        None,
        None,
```

File: rust_src/crates/lisp/src/lisp.rs (hunk 1 above)
Comment (DavidDeSimone) [reply rust_src/crates/lisp/src/lisp.rs:103]: Follow up: It doesn't effect JS, good to go.

**Pull request merged**